fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -454,7 +454,11 @@ public VisibleStateWaiter commitTransaction(long transactionId,
             txnSpan.setAttribute("num_partition", numPartitions);
             unprotectedCommitSpan.end();
             // after state transform
-            transactionState.afterStateTransform(TransactionStatus.COMMITTED, txnOperated, callback, null);
+            try {
+                transactionState.afterStateTransform(TransactionStatus.COMMITTED, txnOperated, callback, null);
```

Comment:
it is possible that unprotectedCommitTransaction() did nothing. but the txnOperated will be set to true. and this `afterStateTransform` will be applied incorrectly.